### PR TITLE
Add base render and action test for item component

### PR DESCRIPTION
**Pull request by tarmann**
For this PR:
- the `title` attribute from the item model
- `markAsRead` action on the component

Thanks for the great talks! Cheers!

### CODE REVIEW

**Comment by tarmann** [comment]
It's based on lorcan's PR. ;)

Next stop: qunit-dom?

File: tests/integration/components/item-component-test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    });
+
+    await setUpAndRender(this, { item });
+
+    assert.equal(find('[data-test-item-title]').innerText, item.get('title'));
+  });
+
+  test('it triggers markAsRead action with item when hide button is clicked', async function(assert){
+    const item = createItem(this.owner);
+    const markAsRead = sinon.spy();
```
Comment (lorcan) [inline tests/integration/components/item-component-test.js]: There's another way of doing this without introducing sinon.  You could try something like:
```
const markAsRead = () => assert.ok(true);
```
YMMV :-)

File: tests/integration/components/item-component-test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    await setUpAndRender(this, {
+      item,
+      markAsRead
+    });
+
+    await click('[data-test-mark-as-read]');
+
+    assert.ok( markAsRead.calledWith( item ) );
```
Comment (lorcan) [inline tests/integration/components/item-component-test.js]: There's an interesting thing happening at this point - we're testing that the `markAsRead` action is being called with `item` but this doesn't actually happen in the component, it happens in the invoker, and in this case on L13 of this file... This means that we're testing the test in a way. 

To overcome this we could add a test to item-list-component where we click on a particular item and ensure that markAsRead is called with that item.

File: tests/integration/components/item-component-test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  test('it renders the content of the item', async function(assert) {
+    const item = createItem(this.owner, {
+      title: 'Hello world'
+    });
+
+    await setUpAndRender(this, { item });
+
+    assert.equal(find('[data-test-item-title]').innerText, item.get('title'));
```
Comment (lorcan) [inline tests/integration/components/item-component-test.js:36]: ❤️

File: tests/integration/components/item-component-test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  test('it triggers markAsRead action with item when hide button is clicked', async function(assert){
+    const item = createItem(this.owner);
+    const markAsRead = sinon.spy();
+
+    await setUpAndRender(this, {
+      item,
+      markAsRead
+    });
+
+    await click('[data-test-mark-as-read]');
```
Comment (lorcan) [inline tests/integration/components/item-component-test.js:48]: ❤️

**Comment by lorcan** [comment]
Looking good, a few small suggestions...

Qunit-dom sounds like a great next step!

File: tests/integration/components/item-component-test.js (hunk 2 above)
Comment (tarmann) [reply tests/integration/components/item-component-test.js]: Makes sense, I'll move this test up.

File: tests/integration/components/item-component-test.js (hunk 1 above)
Comment (tarmann) [reply tests/integration/components/item-component-test.js]: I'm so used to have the assert by the end of the scope for readability that it's quite easy to overlook this kind of optimization.

File: tests/integration/components/item-component-test.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import { click, find, render } from '@ember/test-helpers';
+import { run } from '@ember/runloop';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Component | item-component', function(hooks) {
+  setupRenderingTest(hooks);
+
+  const TEMPLATE = hbs`{{item-component
+    item=item
+    markAsRead=(action markAsRead item)
```
Comment (lorcan) [inline tests/integration/components/item-component-test.js]: we could safely drop `item` from here I think.

**Review by lorcan — APPROVED** [review]
One last suggestion. 

I think we should squash the commits.

Outstanding! 

giphy

**Comment by tarmann** [comment]
Ready to be squashed and merged. 😁

**Comment by lorcan** [comment]
Nice one @tarmann!

**Pull request merged**